### PR TITLE
make compatible w/ Geant4 10.3

### PR DESCRIPTION
**Pull request by gaede**

### CODE REVIEW

File: DDG4/src/Geant4PhysicsConstructor.cpp [hunk 1]
```diff
@@ -35,11 +35,16 @@ namespace  {
     /// Access to a fresh (resetted) instance of the particle table iterator
     G4ParticleTable::G4PTblDicIterator* particleIterator()  const   {
       G4ParticleTable::G4PTblDicIterator* iter;
-#if G4VERSION_NUMBER >= 1000
+
+#if G4VERSION_NUMBER >= 1030
+      iter = GetParticleIterator();
+#else
+ #if G4VERSION_NUMBER >= 1000
```
Comment (andresailer) [inline DDG4/src/Geant4PhysicsConstructor.cpp]: `#elif` maybe?

**Comment by petricm** [comment]
Did you make any tests that that everything works properly with 10.3?

**Comment by gaede** [comment]
So far I only managed to compile against Geant4 10.3 on MacOS Sierra - then I run into problems installing lcgeo ….

Is the current HEAD of DD4hep supposed to build correctly on Sierra w/ the correct rpath setting ?




> On 13 Dec 2016, at 11:34, Marko Petric <notifications@github.com> wrote:
> 
> Did you make any tests that that everything works properly with 10.3?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

------------------------------------ 
Dr. Frank Gaede
DESY - FLC
 Phone:  +49 40 8998 4382    
 Fax:    +49 40 8994 4382
 E-mail: frank.gaede@desy.de

**Comment by peterkostka** [comment]
The macOS Sierra compilation was ok, all test's as well.


Using Geant4-10.3  I get

[ 45%] Building CXX object
DDG4/CMakeFiles/DDG4.dir/src/Geant4PhysicsConstructor.cpp.o
/opt/DD4hep/DD4hep/DDG4/src/Geant4PhysicsConstructor.cpp:39:14: error:
use of undeclared identifier 'aParticleIterator'
      iter = aParticleIterator;
             ^
1 error generated.
make[2]: ***
[DDG4/CMakeFiles/DDG4.dir/src/Geant4PhysicsConstructor.cpp.o] Error 1
make[1]: *** [DDG4/CMakeFiles/DDG4.dir/all] Error 2
make: *** [all] Error 2

which seems to be related to
$GEANT4_DIR/include/Geant4/G4VUserPhysicsList.hh where some definitions
are commented: e.g.    theParticleIterator

**Comment by gaede** [comment]
With this PR it compiles under Sierra w/ G4 10.3 - so I merged it.

**Pull request merged**